### PR TITLE
kv_service: Add a `GetVersion` and `GetMetrics` gRPC endpoint

### PR DESCRIPTION
**Pull request by lance6716**
## What have you changed? (mandatory)

Please explain **IN DETAIL** what the changes are in this PR and why they are needed:
- Summarize your change (mandatory)

Add a `GetVersion` and `GetMetrics` gRPC endpoint as described in #4

`GetVersion` returns version like `3.0.2` and commit hash

`GetMetrics` returns string same with [TiKV rpc](https://github.com/tikv/tikv/blob/master/components/tikv_util/src/metrics/mod.rs#L56,L66)

In order to compile with TiKV, some dependences are updated to not released versions.

- How does this PR work? Need a brief introduction for the changed logic (optional)
- Separately describe each logical change and avoid lazy messages (optional)
- Describe any limitations of the current code (optional)

Please **NOTE** that:
- Do not assume reviewers understand the original issue

## What are the type of the changes? (mandatory)

The currently defined types are listed below, please pick one of the types for this PR by removing the others:
- New feature (change which adds functionality)

## How has this PR been tested? (mandatory)

print them to have a manual check, and add auto-check logic in `tests/intergrations/import/kv_services.rs`

## Does this PR affect TiDB Lightning? (mandatory)
changes the gRPC protocol, thus Yes.

If this PR changes the gRPC protocol (including metrics changes), please file a PR in [tidb-lightning](https://github.com/pingcap/tidb-lightning), and add a reference to here.

[Lightning PR](https://github.com/pingcap/tidb-lightning/pull/226)

## Refer to a related PR or issue link (optional)
[kvproto PR](https://github.com/pingcap/kvproto/pull/436)

### CODE REVIEW

**Comment by lance6716** [comment]
I forgot to signed off at some early commit and found it's hard to solve, so closed #15 and opened this 😢

**Comment by lance6716** [comment]
PTAL @kennytm @leoppro

File: src/import/kv_service.rs [hunk 1]
```diff
@@ -289,4 +289,51 @@ impl ImportKv for ImportKVService {
...
+        _req: GetMetricsRequest,
+        sink: UnarySink<GetMetricsResponse>,
+    ) {
+        let label = "get_metrics";
+        let timer = Instant::now_coarse();
+
+        ctx.spawn(
+            self.threads
+                .spawn_fn(|| {
+                    Ok(metrics::dump())
```
Comment (zier-one) [inline src/import/kv_service.rs]:
```suggestion
                    Ok(dump())
```

Comment (zier-one) [reply src/import/kv_service.rs]: and remove `self` in `use super::metrics::{self, *};`

File: src/import/kv_service.rs [hunk 2]
```diff
@@ -289,4 +289,51 @@ impl ImportKv for ImportKVService {
...
+        ctx.spawn(
+            self.threads
+                .spawn_fn(|| {
+                    Ok((env!("CARGO_PKG_VERSION"), env!("TIKV_BUILD_GIT_HASH")))
+                })
+                .map(|(v, c)| {
+                    let mut res = GetVersionResponse::new();
+                    res.set_version(v.to_owned());
+                    res.set_commit(c.to_owned());
+                    res
```
Comment (kennytm) [inline src/import/kv_service.rs]: Move these inside `spawn_fn`, and remove the `map`.

File: src/import/kv_service.rs [hunk 3]
```diff
@@ -289,4 +289,51 @@ impl ImportKv for ImportKVService {
...
+
+        ctx.spawn(
+            self.threads
+                .spawn_fn(|| {
+                    Ok(metrics::dump())
+                })
+                .map(|v| {
+                    let mut res = GetMetricsResponse::new();
+                    res.set_prometheus(v);
+                    res
```
Comment (kennytm) [inline src/import/kv_service.rs]: (ditto)

File: src/import/metrics.rs [hunk 4]
```diff
@@ -91,3 +91,15 @@ lazy_static! {
         )
         .unwrap();
 }
+
+pub fn dump() -> String {
+    let mut buffer = vec![];
+    let encoder = TextEncoder::new();
+    let metric_familys = prometheus::gather();
```
Comment (kennytm) [inline src/import/metrics.rs]:
```suggestion
    let metric_families = prometheus::gather();
```

File: src/import/kv_service.rs (hunk 1 above)
Comment (kennytm) [reply src/import/kv_service.rs]: In Rust we tend to avoid `use xyz::*`, and `metrics::dump()` is fine.

**Review by kennytm — COMMENTED** [review]
LGTM

**Comment by zier-one** [comment]
LGTM

**Review by zier-one — APPROVED** [review]

**Pull request merged**